### PR TITLE
Perfomance testing (10h) #88

### PR DESCRIPTION
**Pull request by VovaMitsura**

### CODE REVIEW

**Comment by Wectro20** [comment]
|File|Coverage [94.19%]|:green_apple:|
|:-|:-:|:-:|
|[CandidateTimeSlotService.java](https://github.com/Wectro20/intellistart-java-2022-propositum/blob/15379c7b3827a1fc457f113a61a0d45fcacc688f/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservice%2FCandidateTimeSlotService.java)|94.19%|:green_apple:|

|Total Project Coverage|62.29%|:green_apple:|
|:-|:-:|:-:|

File: src/main/java/com/intellias/intellistart/interviewplanning/service/CandidateTimeSlotService.java [hunk 1]
```diff
@@ -128,15 +128,16 @@ private void validateTimeSlot(LocalDate date, LocalTime start, LocalTime end) {
...
-            .findByDateAndEmail(date, candidateEmail);
+        .findByDateAndEmail(date, candidateEmail);
 
     Optional<CandidateTimeSlot> overlappingSlot = slotList
         .stream()
-        .filter(slot -> (start.isAfter(slot.getFrom()) && start.isBefore(slot.getTo()))
-                || (end.isAfter(slot.getFrom()) && end.isBefore(slot.getTo()))
-                || (start.isBefore(slot.getFrom()) && end.isAfter(slot.getTo())))
+        .filter(slot -> (start.equals(slot.getFrom()) && end.equals(slot.getTo()) && date.equals(
+            slot.getDate())) || (start.isAfter(slot.getFrom()) && start.isBefore(slot.getTo()))
```
Comment (Ratchet34702) [inline src/main/java/com/intellias/intellistart/interviewplanning/service/CandidateTimeSlotService.java:138]: No need to check if date equals as we get all slots by email and particular date already

File: src/main/java/com/intellias/intellistart/interviewplanning/service/CandidateTimeSlotService.java [hunk 2]
```diff
@@ -128,15 +128,16 @@ private void validateTimeSlot(LocalDate date, LocalTime start, LocalTime end) {
    * @param end            end time of time slot
    */
```
Comment (Ratchet34702) [inline src/main/java/com/intellias/intellistart/interviewplanning/service/CandidateTimeSlotService.java:129]: This method was changed in main to simplified version with de morgan's law. Can be a merge conflict

**Review by Ratchet34702 — APPROVED** [review]

File: src/main/java/com/intellias/intellistart/interviewplanning/service/CandidateTimeSlotService.java (hunk 1 above)
Comment (VovaMitsura) [reply src/main/java/com/intellias/intellistart/interviewplanning/service/CandidateTimeSlotService.java:138]: Previous validation was`t work properly, so it must be modified in than way

File: src/main/java/com/intellias/intellistart/interviewplanning/service/CandidateTimeSlotService.java (hunk 2 above)
Comment (VovaMitsura) [reply src/main/java/com/intellias/intellistart/interviewplanning/service/CandidateTimeSlotService.java:129]: * Came to decision in tg to save new validation.

**Comment by Wectro20** [comment]
|File|Coverage [94.19%]|:green_apple:|
|:-|:-:|:-:|
|[CandidateTimeSlotService.java](https://github.com/Wectro20/intellistart-java-2022-propositum/blob/3c7a4be059b93e9d4574c8d80053a2a90336dd58/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservice%2FCandidateTimeSlotService.java)|94.19%|:green_apple:|

|Total Project Coverage|61.95%|:green_apple:|
|:-|:-:|:-:|

**Pull request merged**